### DISCUSSION TITLE
How to disable some specific tokenizers?

### DISCUSSION THREAD

**Discussion by ocavue** [micromark/micromark#31] [Q&A] [answered]
# Question: 

How to disable some specific tokenizers when using `micromark` / `mdast-util-from-markdown` / `remark`?

# Why:

I want to use `mdast-util-from-markdown` to parse an **inline** markdown text into some mdast tokens. That means I don't want to use any block-level rule for the text.

My current code is as below:

```typescript
import mdast from "mdast"
import fromMarkdown from "mdast-util-from-markdown"

function parseInlineMarkdown(text: string): mdast.PhrasingContent[] {
    const root: mdast.Root = fromMarkdown(text)
    if (root.children.length === 0) {
        console.warn("root has not children:", root)
        return []
    }

    const content: mdast.Content = root.children[0]
    if (content.type != "paragraph") {
        console.warn("root has not paragraph children:", root)
        return []
    }

    return content.children
}
```

The function will fail when parsing `* hello` since `micromark` will treat it as a list block instead of a paragraph.

It seems that I can fix my issue by somehow disabling all block-level tokenizers (like what [markdown-it](https://github.com/markdown-it/markdown-it/tree/12.0.2#manage-rules) does). But I'm not sure I'm in the right direction. 

# Possible solutions

I found a third party remark plugin that can disable some tokenizers: <https://github.com/zestedesavoir/zmarkdown/tree/master/packages/remark-disable-tokenizers>. However, they haven't migrated to the latest remark version (with `micromark`) so this plugin is currently broken. 

Related issues:

- https://github.com/zestedesavoir/zmarkdown/issues/416
- https://github.com/remarkjs/remark/issues/529

**Comment by wooorm** [comment, marked as answer]
We had https://github.com/remarkjs/remark/issues/552 about this recently, does that help?

**Reply by ocavue** [reply]
Oh, I didn't see this issue. Sorry about that.

It didn't help much, but I think it's better to continue the discussion in one place, so I will close this discussion.
